frappe/commands/site.py

```diff
@@ -123,16 +198,74 @@ def restore(context, sql_file_path, mariadb_root_username=None, mariadb_root_pas
 
 @click.command('partial-restore')
 @click.argument('sql-file-path')
+@click.option('--backup-encryption-key', help='Backup encryption Key')
 @click.option("--verbose", "-v", is_flag=True)
 @pass_context
-def partial_restore(context, sql_file_path, verbose):
+def partial_restore(context, sql_file_path, verbose,  backup_encryption_key=None):
 	from frappe.installer import partial_restore
-	verbose = context.verbose or verbose
+	from frappe.utils.backups import backup_decryption, decryption_rollback
+	
+	if not os.path.exists(sql_file_path):
+		print("Invalid path", sql_file_path)
+		sys.exit(1)
 
+	# Check if file is encrypted 
+	if "enc" in sql_file_path:
+		
+		# Decrypt using the provided key
+		if backup_encryption_key:
+			click.secho(
+				"Encrypted Backup file detected. Decrypting using provided key.", 
+				fg="yellow"
+			)
+			
+			backup_decryption(sql_file_path, backup_encryption_key)
+			
+			# Rollback on unsucessful decryrption
+			if not os.path.exists(sql_file_path):
+				click.secho(
+					"Decryption failed. Please provide a valid key and try again.",
+					fg="red"
+				)
+				decryption_rollback(sql_file_path)
+				sys.exit(1)
+				
+		# Decrypt using the key from site config if key not provided
+		else:
+			click.secho(
+				"Encrypted backup file detected. Decrypting using site config", 
+				fg="yellow"
+			)
+
+			#Get the key from site config
+			site = get_site(context)
+			frappe.init(site)
```

Comment:
Why not do this whole block right before calling `partial_restore`? There doesn't seem to be any reason to initialize werkzeug locals and release them twice here.